### ISSUE TITLE
mvt_alias=Xt overriding vision_alias

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#3148]
`[terrain_type]` supports `aliasof` and also (according to the wiki):
* `def_alias`: like aliasof but overides it for defense calculation only
* `mvt_alias`: like aliasof but overides it for movement calculation only
* `vision_alias`: like aliasof but overides it for vision calculation only

If a terrain has `mvt_alias=Xt` then no value of `vision_alias` will make it see-through for the fog and shroud calculations.

NOT THE MAIN ISSUE: If a terrain has `mvt_alias` but no `vision_alias`, then IIUC the movement is used for the vision. I think that's probably intentional, and the Wiki should say "like aliasof but overrides it for movement and vision only, not defense".

Whether this can be applied to the 1.14 branch: probably not, as it would change the fog and shroud calculations. However, if it only changes terrains that have both mvt_alias and vision_alias then it might affect nothing that's in mainline.

(End of edit)

The graphics  for the rusty iron gate terrain looks as if it should be see-through, but it isn't. The usefulness of this (as well as a workaround) is discussed in #3126.

Looking at the core terrain.cfg in the 1.4 branch, these gates use mvt_alias, suggesting that they're expected to have the vision types from the _bas terrain.

```
[terrain_type]
    ...
    string=^Pr/
    default_base=Rr
    aliasof=_bas,Xt
    mvt_alias=Xt
    ...
[/terrain_type]
```

So I tried making a UMC copy of the terrain, adding `vision_alias=_bas`, `vision_alias=Urb` and `vision_alias=+,Urb` and changing `aliasof` to `aliasof=_bas` (on a base terrain of Urb or Rr).

All of the above still blocked vision, until I changed the `mvt_alias` to be `mvt_alias=_bas`.

**Comment by stevecotton** [comment]
I think the "Graphics" tag doesn't fit this bug, this one is about which hexes on the map have fog or shroud.

If the decision is to keep this terrain blocking vision for shroud/fog calculations, then maybe a separate bug (or just a PR) for changing these gate's graphics to a solid-iron-gate would be needed.

**Comment by Vultraz** [comment]
I don't think it's necessary to have these terrains be see-through simply because they have bars in their graphics. It make sense as a gameplay abstraction that a gate in a scenario is a barrier beyond which units cannot see until it is removed, especially since that's how gates have been deployed in various UMC, even though the old graphics often used for gates also had bars. There's also the additional matter of the wooden vs metal gates. Would the metal gates be see-through, but the wooden not? This just seems like an over-complication, and would break a lot of scenarios that treat gates as impervious.

**Comment by Vultraz** [comment]
Is it a bug? I can't really tell. I thought it was more an FR?

**Comment by CelticMinstrel** [comment]
I'm not sure? But my point is that your statement about the gates isn't necessarily a reason to close this.

**Comment by Vultraz** [comment]
Fair enough. This can be reopened if there is indeed a bug.

**Comment by stevecotton** [comment]
A cause of this bug, but maybe not the only one: movetype.cpp#L287-L289 only supports the mvt_alias and def_alias.

https://github.com/wesnoth/wesnoth/blob/211f14176ce57509003ca4542d388c526157093c/src/movetype.cpp#L287-L289

**Comment by jostephd** [comment]
What is this issue about? There seem to be at least three difference matters here:

- "If a terrain has mvt_alias=Xt then no value of vision_alias will make it see-through for the fog and shroud calculations."
- "The graphics for the rusty iron gate terrain looks as if it should be see-through, but it isn't."
- "If a terrain has mvt_alias but no vision_alias, then IIUC the movement is used for the vision."

**Comment by stevecotton** [comment]
> What is this issue about?

1. This
2. Not this
3. This is part of getting to the root cause of 1.

**Comment by stevecotton** [comment]
The wiki has already been updated, and I've just merged #4278 to give a warning that vision_alias is not supported.
